### DISCUSSION TITLE
Skip mutants based on custom headers

### DISCUSSION THREAD

**Discussion by ASuciuX** [sourcefrog/cargo-mutants#351] [Ideas]
Hey @sourcefrog , there are a lot of tests on stacks-core that are not flagged as `test`, but instead have macros for that like `#[apply(test_clarity_versions)]`, reference here https://github.com/stacks-network/stacks-core/blob/8235472058e1a981f3badb7e897e418281a80ec5/clarity/src/vm/tests/variables.rs#L31.

Is there a method to add headers, besides `#[test]`, to be directly skipped and not generate mutants?  Would be very useful as it would keep the code cleaner and be directly automatic instead of creating skip headers for each of them.  If there isn't could something like this be implemented? Thank you very much.

**Comment by sourcefrog** [comment]
I think if you put `#cfg[test]` on that module it will be skipped by mutants. Does that not work?

**Reply by ASuciuX** [reply]
That's right, that way it works, but there are a lot of this type of functions and we wouldn't want to add the header for every single one. Instead, if we could parse to the .cargo/config that it should also skip this header `#[apply(test_clarity_versions)]`, it would be very useful and wouldn't require any manual effort when new ones are created/modified.

**Reply by sourcefrog** [reply]
I was saying you could put it on the whole module, not each individual test.

Now I can see the module tests does actually have an attribute like this

https://github.com/stacks-network/stacks-core/blob/8235472058e1a981f3badb7e897e418281a80ec5/clarity/src/vm/mod.rs#L44-L45

```rust
#[cfg(any(test, feature = "testing"))]
pub mod tests;
```

but cargo-mutants does not yet understand the `any` boolean operator, which is #50.

I think it would work to put `#[mutants::skip]` on the `mod test`. Could you try that? (You don't need to run all the tests, just `cargo mutants --list` will show you whether it worked.)

Another option is to put `exclude_globs = tests` into `.cargo/mutants.toml` if you never want to mutate code in directories called `tests` which seems likely. But it would be good to confirm the first one works too.

**Reply by ASuciuX** [reply]
Thanks, not at the computer currently, I'll come with feedback.

**Reply by ASuciuX** [reply]
Putting the `#[mutants::skip]` annotation on the `mod test` worked, thank you very much!
